pkg/compact/schedule_delete.go

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package compact
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/oklog/ulid"
+	"github.com/pkg/errors"
+	"github.com/thanos-io/thanos/pkg/block"
+	"github.com/thanos-io/thanos/pkg/objstore"
+	"github.com/thanos-io/thanos/pkg/runutil"
+)
+
+// ScheduleDeleteFilename is file to store compactor metadata
+// about when the block is scheduled to be deleted.
+const ScheduleDeleteFilename = "compactor-meta.json"
+
+// CompactorMeta stores block id and when block was marked for deletion.
+type CompactorMeta struct {
+	ID           ulid.ULID `json:"id"`
+	DeletionTime int64     `json:"deletion_time"`
+}
+
+// ScheduleBlockDelete marks the block to be deleted.
+type ScheduleBlockDelete struct {
+	dir         string
+	logger      log.Logger
+	deleteDelay time.Duration
+	bkt         objstore.Bucket
+}
+
+// NewScheduleBlockDelete creates a new ScheduleBlockDelete.
+func NewScheduleBlockDelete(logger log.Logger, dir string, bkt objstore.Bucket, deleteDelay time.Duration) *ScheduleBlockDelete {
+	return &ScheduleBlockDelete{
+		dir:         dir,
+		logger:      logger,
+		deleteDelay: deleteDelay,
+		bkt:         bkt,
+	}
+}
+
+// ScheduleDelete deletes blocks from bucket
+// deleteDelay duration after block is marked for deletion.
+func (s *ScheduleBlockDelete) ScheduleDelete(ctx context.Context) error {
+	return filepath.Walk(s.dir, func(path string, info os.FileInfo, err error) error {
```

Comment:
What is this? Why we walk over local disk not bucket? :thinking:

Reply:
Right now I am uploading this file to storage